### ISSUE TITLE
about VAE (why should we save the image?)

### ISSUE THREAD

**Issue by JingsenZhang** [yunjey/pytorch-tutorial#201]
I don't understand when doing training why we should save the sampled image and reconstructed image in every epoch.
Just like this.

with torch.no_grad():
        # Save the sampled images
        z = torch.randn(batch_size, z_dim).to(device)
        out = model.decode(z).view(-1, 1, 28, 28)
        save_image(out, os.path.join(sample_dir, 'sampled-{}.png'.format(epoch+1)))

        # Save the reconstructed images
        out, _, _ = model(x)
        x_concat = torch.cat([x.view(-1, 1, 28, 28), out.view(-1, 1, 28, 28)], dim=3)
        save_image(x_concat, os.path.join(sample_dir, 'reconst-{}.png'.format(epoch+1)))

AND  I have tried the code without the part above,There's not much of a difference about the results.

who can help me?  Thank you very very much!

**Comment by OrangeC93** [comment]
Just save interim output for people to see and compare? 
I don't think there're any influence on the model performance.

**Comment by JingsenZhang** [comment]
I got it.  


Thank you very much！


| |
Jason
|
|
邮箱：18742521286@163.com
|

Signature is customized by Netease Mail Master

On 12/10/2019 06:29, OrangeC93 wrote:

Just save interim output for people to see and compare?
I don't think there're any influence on the model performance.
